client/src/components/Header/components/login.tsx

```diff
@@ -37,10 +37,16 @@ const Login = ({
       data-test-label={dataTestLabel}
       data-playwright-test-label='header-sign-in-button'
       href={href}
-      onClick={() => {
+      onClick={(e: React.MouseEvent<HTMLAnchorElement>) => {
+        // Prevent default to make sure GA is called before the redirect
+        e?.preventDefault();
         callGA({
           event: 'sign_in'
         });
+        // Ensure async operations are completed
+        setTimeout(() => {
+          window.location.href = href;
+        }, 0); //
```

Comment:
I think this is worse. If you really _just_ want to know when the button is clicked, then what you had was fine.

My previous point was, Campers could click the button, fail to be signed in, and you will still get _x number of sign_in events_.